captum/attr/_core/feature_permutation.py

```diff
@@ -24,27 +24,47 @@ def _permute_feature(x: Tensor, feature_mask: Tensor) -> Tensor:
 
 class FeaturePermutation(FeatureAblation):
     r"""
-    This attribution method essentially implements the permutation feature
-    importance algorithm, as described here:
-    https://christophm.github.io/interpretable-ml-book/feature-importance.html
+    A perturbation based approach to compute attribution, which
+    takes each input feature, permutes the feature values within a batch,
+    and computes the difference in output between the original and shuffled.
 
-    A basic tl;dr of the algorithm is:
+    Example pseudocode for the algorithm is as follows::
 
-    perm_feature_importance(batch):
-        importance = dict()
-        baseline_error = error_metric(model(batch), batch_labels)
-        for each feature:
-            permute this feature across the batch
-            error = error_metric(model(permuted_batch), batch_labels)
-            importance[feature] = baseline_error - error
-            "un-permute" the feature across the batch
+        perm_feature_importance(batch):
+            importance = dict()
+            baseline_error = error_metric(model(batch), batch_labels)
+            for each feature:
+                permute this feature across the batch
+                error = error_metric(model(permuted_batch), batch_labels)
+                importance[feature] = baseline_error - error
+                "un-permute" the feature across the batch
 
-        return importance
+            return importance
 
     It should be noted that the `error_metric` must be called in the
-    `forward_func`. You do not need to provide an error metric, e.g. you
+    `forward_func`. You do not need to have an error metric, e.g. you
     could simply return the logits (the model output), but this may or may
     not provide a meaningful attribution.
+
+    This method, unlike other attribution methods, requires a batch
+    of examples to compute attributions and cannot be performed on a single example.
+
+    By default, each scalar value within
+    each input tensor is taken as a feature and shuffled independently. Passing
+    a feature mask, allows grouping features to be shuffled together.
+    Each input scalar in the group will be given the same attribution value
+    equal to the change in target as a result of shuffling the entire feature
+    group.
+
+    The forward function can either return a scalar per example, or a single
+    scalar for the full batch. If a single scalar is returned for the batch,
+    `perturbations_per_eval` must be 1, and the returned attributions will have
+    first dimension 1, corresponding to feature importance across all
```

Comment:
nit: what is the features are structured multi-dimensionally ? Maybe we don't want to say `first dimension 1` ?

Reply:
I think it still would have first dimension 1? It's just that the attributions would be the size of one input example.